core/controlplane/config/templates/cloud-config-controller

```diff
@@ -3092,20 +3204,32 @@ write_files:
           fi
       done
 
+  {{if .Experimental.AuditLog.Enabled -}}
+  # Check worker communication by searching audit logs
   - path: /opt/bin/check-worker-communication
     permissions: 0700
     owner: root:root
     content: |
       #!/bin/bash -e
       set -ue
 
+      AUDIT_LOG_PATH="{{.Experimental.AuditLog.LogPath}}"
+
       kubectl() {
         /usr/bin/docker run --rm --net=host -v /srv/kubernetes:/srv/kubernetes {{.HyperkubeImage.RepoWithTag}} /hyperkube kubectl "$@"
       }
 
       queryserver() {
         echo "Checking to see if workers are communicating with API server."
-        docker logs --tail 10 ${DOCKERIMAGE} |& grep -v 127.0.0.1 | awk 'BEGIN {C=1} /kubelet\// && $8 == "200" {C=0} END {exit C}'
+        auditlogs | grep -v 127.0.0.1 | awk 'BEGIN {C=1} /kubelet\// && $8 == "200" {C=0} END {exit C}'
+      }
+
+      auditlogs() {
+        if [ "$AUDIT_LOG_PATH" == "/dev/stdout" ]; then
+          docker logs --since 11s ${DOCKERIMAGE} |& cat
```

Comment:
Is there a reason behind this magic number here (`11s`)?

Reply:
`queryserver` is called every 10+ seconds below. I wanted to gather logs for periods slightly longer than that not to drop any lines(hopefully)